rest/src/market_data_rest.py

```diff
@@ -0,0 +1,232 @@
+import json
+import time
+from abc import ABC, abstractmethod
+from http import HTTPStatus
+
+import requests
+
+from rest.src.api_utils import get_header_key_col, get_header_signature_col, get_api_url
+from rest.src.request_types import GET, make_request
+
+
+class MarketDataRestApi(ABC):
+    """
+    A base class for all Market Data REST APIs
+
+    When adding a method that should be enforced on all children, use @abstractmethod
+    """
+
+    def __init__(self,
+                 market,
+                 instrument_type,
+                 public_path='',
+                 private_path='',
+                 public_key=None,
+                 private_key=None):
+        """
+
+        :param market:
+        :param instrument_type:
+        :param public_path:
+        :param private_path:
+        :param public_key:
+        :param private_key:
+        """
+        self.market = market
+        self.instrument_type = instrument_type
+        self.session = requests.Session()
+        # The base url of the API depends on the market and instrument type
+        self.api_url = get_api_url(market, instrument_type)
+
+        # When doing private queries, the name of the header columns used to pass the signature and API key
+        self.header_key_col = get_header_key_col(market, instrument_type)
+        self.header_signature_col = get_header_signature_col(market, instrument_type)
+
+        # Optional, some APIs have 2 different routes for public and private querues
```

Comment:
"Optional, some APIs have 2 different routes for public and private **queries**"

Reply:
Fixed